src/tools/compiletest/src/runtest.rs

```diff
@@ -2560,8 +2560,15 @@ impl<'test> TestCx<'test> {
             Assembly | Codegen => {
                 rustc.arg("-Cdebug-assertions=no");
             }
+            Crashes => {
+                let mir_dump_dir = self.get_mir_dump_dir();
+                remove_and_create_dir_all(&mir_dump_dir);
+                let mut dir_opt = "-Zdump-mir-dir=".to_string();
+                dir_opt.push_str(mir_dump_dir.to_str().unwrap());
+                rustc.arg(dir_opt);
```

Comment:
I wouldn't mind some abstraction. Could we even just stuff this in a closure and call it on both arms that need this?